core/test/test_container.cpp

```diff
@@ -685,3 +686,52 @@ TEST(Fallback, failing) {
 	EXPECT_FALSE(t.plan());
 	EXPECT_EQ(t.solutions().size(), 0u);
 }
+
+TEST(Fallback, ConnectStageInsideFallbacks) {
+	resetMockupIds();
+	Task t;
+	t.setRobotModel(getModel());
+
+	t.add(std::make_unique<GeneratorMockup>());
+
+	auto fallbacks = std::make_unique<Fallbacks>("Fallbacks");
+	fallbacks->add(std::make_unique<ConnectMockup>());
+	t.add(std::move(fallbacks));
+
+	t.add(std::make_unique<GeneratorMockup>());
+
+	EXPECT_TRUE(t.plan());
+	EXPECT_EQ(t.numSolutions(), 1u);
+}
+
+TEST(Fallback, ComputeFirstSuccessfulStageOnly) {
+	resetMockupIds();
+	Task t;
+	t.setRobotModel(getModel());
+
+	t.add(std::make_unique<GeneratorMockup>());
+
+	auto fallbacks = std::make_unique<Fallbacks>("Fallbacks");
+	fallbacks->add(std::make_unique<ForwardMockup>(PredefinedCosts::constant(0.0)));
+	fallbacks->add(std::make_unique<ForwardMockup>(PredefinedCosts::constant(0.0)));
+	t.add(std::move(fallbacks));
+
+	EXPECT_TRUE(t.plan());
+	EXPECT_EQ(t.numSolutions(), 1u);
+}
+
+TEST(Fallback, ActiveChildReset) {
+	resetMockupIds();
+	Task t;
+	t.setRobotModel(getModel());
+
+	t.add(std::make_unique<GeneratorMockup>(PredefinedCosts{ { 0.0, INF, 0.0 } }));
+
+	auto fallbacks = std::make_unique<Fallbacks>("Fallbacks");
+	fallbacks->add(std::make_unique<ForwardMockup>(PredefinedCosts::constant(0.0)));
+	fallbacks->add(std::make_unique<ForwardMockup>(PredefinedCosts::constant(0.0)));
+	t.add(std::move(fallbacks));
+
+	EXPECT_TRUE(t.plan());
+	EXPECT_EQ(t.numSolutions(), 4u);
```

Comment:
Depending on the preferred semantics (whether fallback alternatives should be computed at all), the test is still not correct: If no alternatives should be considered you would have to check for *two* solutions where both should come from the first `ForwardMockup`.

Let's wait for Robert's response before modifying this.

Reply:
I don't yet fully understand the purpose of this test. I would expect two solutions here, contributed by the _first_ child. However, this was not yet tested here. I pushed a corresponding fixup.

Reply:
There should be another test case, where the external generator is failing first before succeeding.
Currently, I would expect this test to fail (with a solution of the wrong, second child).

Reply:
The test suggested that after computing solutions for the second spawned solution from the generator, the `active_child_` would not be reset to the first child for computing the third solution. Before your two commits, this led to the first child being ignored for the third spawned generator solution.

As @v4hn pointed out, this test was not consistent with the proposed functionality in `ComputeFirstSuccessfulStageOnly`, but rather tried to test the status quo. But even after your fix for the `ActiveChildReset` test, I would have expected the test to fail, because as I understand it, the `active_child_` is pointing to the second child after the second spawned generator solution. So for the third solution, I expected the container to use the second child for computation. However, the test passes, so it is actually using the first child. I'm not sure how your commit fixed this.

I just tried adding another test as you suggested, where the external generator is failing first before succeeding:
```cpp
std::make_unique<GeneratorMockup>(PredefinedCosts{ { INF, 0.0 } }));
```
However, this yields 0 solutions and I'm not sure why.